### PR TITLE
stability: add scale & upgrade case functions

### PR DESCRIPTION
**Pull request by zyguan**
I just implement two case functions for `OperationActions`, PTAL.

### CODE REVIEW

File: tests/actions.go [hunk 1]
```diff
@@ -333,6 +361,16 @@ func (oa *operatorActions) Restore(from *TidbClusterInfo, jobName string, to *Ti
 func (oa *operatorActions) DeployMonitor(info *TidbClusterInfo) error { return nil }
 func (oa *operatorActions) CleanMonitor(info *TidbClusterInfo) error  { return nil }
 
+func getComponentContainer(set *v1beta1.StatefulSet) (corev1.Container, bool) {
+	name := set.Labels["app.kubernetes.io/component"]
```
Comment (weekface) [inline tests/actions.go]: use `label.ComponentLabelKey` instead

File: tests/actions.go [hunk 2]
```diff
@@ -349,10 +353,44 @@ func (oa *operatorActions) StopInsertDataTo(info *TidbClusterInfo) error {
...
+		info.ClusterName, chartPath("tidb-cluster", info.OperatorTag), info.HelmSetString())
+	glog.Info("[SCALE] " + cmd)
+	res, err := exec.Command("/bin/sh", "-c", cmd).CombinedOutput()
+	if err != nil {
+		return errors.Wrapf(err, "failed to scale tidb cluster: %s", string(res))
+	}
+	return nil
+}
+
+func (oa *operatorActions) UpgradeTidbCluster(info *TidbClusterInfo) error {
```
Comment (xiaojingchen) [inline tests/actions.go]: change to `UpgradeTidbClusterTo(info *TidbClusterInfo,version string)`

**Comment by weekface** [comment]
Add it the the main func to ensure it works as expected.

File: tests/actions.go [hunk 3]
```diff
@@ -349,10 +353,44 @@ func (oa *operatorActions) StopInsertDataTo(info *TidbClusterInfo) error {
...
 
-func (oa *operatorActions) ScaleTidbCluster(info *TidbClusterInfo) error   { return nil }
-func (oa *operatorActions) UpgradeTidbCluster(info *TidbClusterInfo) error { return nil }
-func (oa *operatorActions) DeployMonitor(info *TidbClusterInfo) error      { return nil }
-func (oa *operatorActions) CleanMonitor(info *TidbClusterInfo) error       { return nil }
+func chartPath(name string, tag string) string {
+	return "/charts/" + tag + "/" + name
+}
+
+func (oa *operatorActions) ScaleTidbCluster(info *TidbClusterInfo) error {
```
Comment (xiaojingchen) [inline tests/actions.go]: change to `ScaleTidbClusterTo(info *TidbClusterInfo,pdReplicas int,tikvReplicas int,tidbReplicas int)` is more reasonable

**Comment by xiaojingchen** [comment]
the scale and upgrade method have not contained check logic code

File: tests/actions.go (hunk 2 above)
Comment (zyguan) [reply tests/actions.go]: But how to upgrade only one component (eg. just upgrade tidb) ?

**Comment by zyguan** [comment]
> Add it the the main func to ensure it works as expected.

I'm working on it, however check for an upgrade often timeout.

> each test case consists of two parts, triggering the event first, and then checking for success.the scale and upgrade method have not contained checking logic.

I thought `CheckTidbClusterStatus` is an all-in-one check method, should we add checks to `OperatorActions` for each operation ? @weekface

File: tests/cmd/e2e/main.go [hunk 4]
```diff
@@ -68,18 +70,35 @@ func main() {
...
+			"tidb.resources.requests.memory": "1Gi",
+		},
+		Args: map[string]string{},
 	}
 
+	perror(oa.CleanTidbCluster(clusterInfo))
+	perror(oa.DeployTidbCluster(clusterInfo))
+	perror(oa.CheckTidbClusterStatus(clusterInfo))
+
+	clusterInfo = clusterInfo.ScaleTiDB(3)
```
Comment (weekface) [inline tests/cmd/e2e/main.go]: We should **scale in** and **scale out** all the components: pd, tikv and tidb.

**Comment by weekface** [comment]
@zyguan For scale and upgrade, there are many other checks that need to do other than the basic checks, for details you can see [here](https://github.com/pingcap/schrodinger-test/blob/operator-test/operator/pkg/cluster/operator.go#L128-L297) and [here](https://github.com/pingcap/tidb-operator/blob/master/tests/e2e/scale.go) and [here](https://github.com/pingcap/tidb-operator/blob/master/tests/e2e/upgrade.go).

But for the first commit, i think this PR is OK if you add more scale and uprade case: https://github.com/pingcap/tidb-operator/pull/309#discussion_r265558788 .

**Comment by weekface** [comment]
@zyguan For scale and upgrade, there are many other checks that need to do other than the basic checks, for details you can see [here](https://github.com/pingcap/schrodinger-test/blob/operator-test/operator/pkg/cluster/operator.go#L128-L297) and [here](https://github.com/pingcap/tidb-operator/blob/master/tests/e2e/scale.go) and [here](https://github.com/pingcap/tidb-operator/blob/master/tests/e2e/upgrade.go).

But for the first commit, i think this PR is OK if you add more scale and uprade case: https://github.com/pingcap/tidb-operator/pull/309#discussion_r265558788 .

File: tests/cmd/e2e/main.go (hunk 4 above)
Comment (weekface) [reply tests/cmd/e2e/main.go]: Yes, you can do this in another PR.

**Comment by zyguan** [comment]
> @zyguan For scale and upgrade, there are many other checks that need to do other than the basic checks, for details you can see [here](https://github.com/pingcap/schrodinger-test/blob/operator-test/operator/pkg/cluster/operator.go#L128-L297) and [here](https://github.com/pingcap/tidb-operator/blob/master/tests/e2e/scale.go) and [here](https://github.com/pingcap/tidb-operator/blob/master/tests/e2e/upgrade.go).
> 
> But for the first commit, i think this PR is ok to merging.

I'm already working on it, also, I'm trying to port ddl test as a workload. I'll commit these works in next few PRs.

**Comment by weekface** [comment]
Great!

**Pull request merged**